### PR TITLE
adding-mix-panel-offline-support

### PR DESCRIPTION
**Pull request by Ajeey**

### CODE REVIEW

**Review by mehtayash — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -22,6 +25,28 @@ export default function mixpanel({
...
+    // When network is back and actions are available in the queue, emit mixpanel event for each action
+    if (state.network.available && arr.length > 0) {
+      arr.forEach((x, i) => {
+        trackEvent({
+          token,
+          distinctId,
+          eventName: eventName,
+          eventData: properties
+        })
+      })
```
Comment (pgodha) [inline src/index.js:46]: @sujithbluejeans where are we referring x here?

File: src/index.js [hunk 2]
```diff
@@ -22,6 +25,28 @@ export default function mixpanel({
...
+      arr.splice(0, arr.length)
+    }
```
Comment (pgodha) [inline src/index.js:48]: So if I understand it correctly, when we want to fire an event we check network, if is not available then we push data into arr, and next time when network it available then  we check arr and if it is not empty then we fire one by one previous events. What if next event is getting fired after a long time, or never fired, im that case offline events will never get fired.

Comment (sujithbluejeans) [reply src/index.js:48]: @pgodha This is the scenario when network is down for the entire session the user is in.

Comment (sujithbluejeans) [reply src/index.js:48]: Here is the requirement for the mix-panel task.
**Both Fiber and App will send many mixpanel events when network is down. If network connectivity is lost the mixpanel events should not get lost or ignored. They should be sent once network connectivity is back.**

File: src/index.js (hunk 1 above)
Comment (sujithbluejeans) [reply src/index.js:46]: @pgodha Thanks for pointing Pranay. I overlooked this error while copy pasting from node modules to this repo.

File: src/index.js (hunk 2 above)
Comment (pgodha) [reply src/index.js:48]: But here we are not sending events once network is up. Instead we are waiting for next event and then we are sending all previous events.

Comment (sujithbluejeans) [reply src/index.js:48]: Once the network is back, we will be sending a mix-panel event for sure Prany. Or else what we can  do is emit a mix-panel event once the network is back, to make sure we send the offline queue events.

Comment (pgodha) [reply src/index.js:48]: That would work for now.

Comment (sujithbluejeans) [reply src/index.js:48]: @pgodha We can merge and publish this in that case Pranay.

**Review by pgodha — APPROVED** [review]

**Pull request merged**